wgpu-types/src/instance.rs

```diff
@@ -35,6 +35,12 @@ pub struct InstanceDescriptor {
     pub memory_budget_thresholds: MemoryBudgetThresholds,
     /// Options the control the behavior of specific backends.
     pub backend_options: crate::BackendOptions,
+    /// System platform or compositor connection to connect this `Instance` to.
+    ///
+    /// - On GLES, this is required when intending to present on the platform.
+    /// - On Vulkan, this could be used to skip extensions.
+    // Cannot have the safe borrow variant here...
```

Comment:
Why not? We could deal with the unsafety if we absolutely have to, but with `RWH` having a safe variant, it would be ideal if we could make the default path safe.

Reply:
I just added a lifetime parameter to `wgpu_types::InstanceDescriptor` (which was only on `wgpu_hal::InstanceDescriptor` previously) and that works fine.